### ISSUE TITLE
Indentation check doesn't give violation when google formatter does

### ISSUE THREAD

**Issue by rdiachenko** [checkstyle/checkstyle#8965]
Checkstyle 8.37, google-java-format 1.9
config (taken from google_checks.xml - https://github.com/checkstyle/checkstyle/blob/checkstyle-8.37/src/main/resources/google_checks.xml#L243):
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>
```
code:
```java
class Test {
  public Object foo() {
    return Optional.empty()
        .orElseThrow(
          () ->
new IllegalArgumentException(
"Something wrong 1, something wrong 2, something wrong 3"));
  }
}
```
none violations given by Checkstyle:
```bash
$ java -jar checkstyle-8.37-all.jar -c config.xml Test.java 
Starting audit...
Audit done.
```
but there is a non-empty diff when applying google-java-format (which leads to errors when using https://github.com/coveooss/fmt-maven-plugin):
```bash
$ java -jar google-java-format-1.9-all-deps.jar Test.java > /tmp/FormattedTest.java && diff -Naru Test.java /tmp/FormattedTest.java
--- Test.java	2020-11-05 06:46:28.027335639 +0000
+++ /tmp/FormattedTest.java	2020-11-05 07:13:01.536163532 +0000
@@ -2,8 +2,8 @@
   public Object foo() {
     return Optional.empty()
         .orElseThrow(
-          () ->
-new IllegalArgumentException(
-"Something wrong 1, something wrong 2, something wrong 3"));
+            () ->
+                new IllegalArgumentException(
+                    "Something wrong 1, something wrong 2, something wrong 3"));
   }
 }
```

**Comment by aryaniiit002** [comment]
if i may ask what is 
> $ java -jar google-java-format-1.9-all-deps.jar Test.java > /tmp/FormattedTest.java && diff -Naru Test.java /tmp/FormattedTest.java

this command and what is this  `/tmp/FormattedTest.java`?

**Comment by rdiachenko** [comment]
@aryaniiit002 
> if i may ask what is this command and what is this /tmp/FormattedTest.java?

* `java -jar google-java-format-1.9-all-deps.jar Test.java > /tmp/FormattedTest.java` - format `Test.java` using google-java-format and output the formatted code into `/tmp/FormattedTest.java`
* `diff -Naru Test.java /tmp/FormattedTest.java` - find a difference between the original and formatted code

However, I don't understand the second part of your comment. Could you clarify please what was the question?

**Comment by aryaniiit002** [comment]
the checkstyle output i am getting is not same as you are getting.

**Comment by aryaniiit002** [comment]
thank you for that 

> Could you clarify...


the checkstyle output i am getting is not same as you are getting.

**Comment by Abhishek-kumar09** [comment]
@rdiachenko 
The indentation problems related to new keyword is fixed in https://github.com/checkstyle/checkstyle/pull/8983
And you are getting error logs with 8.41
<details><summary>checkstyle output with 8.41</summary>
<p>

```java
class Test {
  public Object foo() {
    return Optional.empty()
        .orElseThrow(
          () ->
new IllegalArgumentException(
"Something wrong 1, something wrong 2, something wrong 3"));
  }
}
```
config:
```xml
 <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="2"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
```

$ java -jar checkstyle-8.41-all.jar -c config.xml t2.java 
```
Starting audit...
[ERROR] /home/abhishhek/playground/t2.java:6:1: 'new' has incorrect indentation level 0, expected level should be one of the following: 10, 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

</p>
</details>

But still there is some problem: 
$ t2.java
```java
class Test {
  public Object foo() {
    return Optional.empty()
        .orElseThrow(
          () ->
            new IllegalArgumentException(
"Something wrong 1, something wrong 2, something wrong 3")); // expected a violation here
  }
}
```

**Actual**
java -jar checkstyle-8.41-all.jar -c config.xml t2.java 
```
Starting audit...
Audit done.
```

**expected**
violation on line 7

**Comment by MUzairS15** [comment]
```
src % cat config.xml Test.java                                 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="IndentationCheck">
        </module>
    </module>
    <module name="SuppressionSingleFilter">
        <property name="lines" value="5"/>
    </module>
</module>

import java.util.function.Function;

enum Test {
        t1(
           v ->
v, // violation expected
        new Object()
),
        t2(
v -> v, // violation expected and reported
        new Object()
        );

    Test(Function<Object, Object> function, Object o1) {
    }
}

src % java -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[WARN] /Users/shabana/Documents/OpenSource/CheckStyleTest/src/src/Test.java:10:1: 'lambda arguments' has incorrect indentation level 0, expected level should be 4. [Indentation]
Audit done.

```

Suppression is used to prevent existing issue #11103,
Check fails to recognize when line wrapping is used
